File: Creational/AbstractFactory/C++/products/WinCheckbox.cpp
#include <iostream>
#include "WinCheckbox.h"

/**
 * Concrette object of family (realisation).
 */

void WinCheckbox::paint()
{
	std::cout << "Painting checkbox in Windows style." << std::endl;
}

void WinCheckbox::setPosition(const int x, const int y)
{
	std::cout << "Setting checkbox position to (" << x << ", " << y << ") in Windows style." << std::endl;
}

bool WinCheckbox::getValue()
{
	std::cout << "Getting checkbox value in Windows style." << std::endl;
	return false;
}

File: Creational/AbstractFactory/PHP/products/WinCheckbox.php
<?php

require_once(__DIR__.'/../interfaces/ICheckbox.php');

/**
 * Concrette object of family.
 */

class WinCheckbox implements ICheckbox
{
	/**
	 * @override
	 */
	public function paint()
	{
		print("Painting checkbox in Windows style.\n");
	}

	/**
	 * @override
	 */
	public function setPosition($x, $y)
	{
		print("Setting checkbox position to ({$x}, {$y}) in Windows style.\n");
	}

	/**
	 * @override
	 */
	public function getValue()
	{
		print("Getting checkbox value in Windows style.\n");
		return false;
	}
}
